src/mds/nameserver2/curvefs.cpp

```diff
@@ -280,6 +282,25 @@ StatusCode CurveFS::CreateFile(const std::string & fileName,
         }
     }
 
+    std::vector<PoolIdType> logicalPools =
+                topology_->GetLogicalPoolInCluster();
+    std::map<PoolIdType, double> enoughSpace;
+    bool hasEnough = false;
+    chunkSegAllocator_->GetRemainingSpaceInLogicalPool(&logicalPools,
+                                &enoughSpace);
+
+    for (auto it = enoughSpace.begin(); it != enoughSpace.end(); it++) {
```

Comment:
A file can be created as long as the accumulated value of all logical pools can exceed length.